project-management/src/main/java/life/qbic/projectmanagement/domain/service/SampleDomainService.java

```diff
@@ -70,25 +92,30 @@ public void updateSamples(Project project, Collection<SampleUpdateRequest> updat
           sampleInfo.sampleInformation().specimen(), sampleInfo.sampleInformation().analyte()));
       sample.setComment(sampleInfo.sampleInformation().comment());
       sample.setExperimentalGroupId(sampleInfo.sampleInformation().experimentalGroup().id());
+      sample.update(sampleInfo);
     }
     sampleRepository.updateAll(project, samplesToUpdate);
+    domainEventsCache.forEach(
+        domainEvent -> DomainEventDispatcher.instance().dispatch(domainEvent));
   }
 
   public void deleteSamples(Project project, BatchId batchId, Collection<SampleId> samples) {
     Objects.requireNonNull(samples);
     sampleRepository.deleteAll(project, samples);
     samples.forEach(sampleId -> dispatchSuccessfulSampleDeletion(sampleId, batchId));
+    if(!samples.isEmpty()) {
+      dispatchProjectChangedUponSampleDeletion(project.getId());
+    }
   }
 
   private void dispatchSuccessfulSampleDeletion(SampleId sampleId, BatchId batchId) {
     SampleDeleted sampleDeleted = SampleDeleted.create(batchId, sampleId);
     DomainEventDispatcher.instance().dispatch(sampleDeleted);
   }
 
-  private void dispatchSuccessfulSampleRegistration(Sample sample) {
-    SampleRegistered sampleRegistered = SampleRegistered.create(sample.assignedBatch(),
-        sample.sampleId());
-    DomainEventDispatcher.instance().dispatch(sampleRegistered);
+  private void dispatchProjectChangedUponSampleDeletion(ProjectId projectId) {
+    ProjectChanged projectChanged = ProjectChanged.create(projectId);
+    DomainEventDispatcher.instance().dispatch(projectChanged);
```

Comment:
where did the sample registered event go to?

Reply:
this is the delete event and it cannot be handled inside the Sample object, so @sven1103 and I agreed that we can dispatch a project changed event here, instead.

Reply:
yes but the sample registered event vanished on your branch. Maybe I am missing something?

Reply:
The creation and update events are in the Sample (etc.) classes themselves, now. in the SampleDomainService these events are cached and then forwarded, if registration/update succeeds. In this case in the registerSamples() method.